cmd/legacy_param_mapper_test.go

```diff
@@ -451,3 +451,25 @@ func TestCustomEndpointResolutionFromFlags(t *testing.T) {
 		assert.Equal(t, *resolvedConfig.GcsConnection.CustomEndpoint, *u)
 	}
 }
+
+func TestValidClientProtocol(t *testing.T) {
+	flags := &flagStorage{
+		ClientProtocol: mountpkg.ClientProtocol("http1"),
+	}
+
+	v, err := PopulateNewConfigFromLegacyFlagsAndConfig(&mockCLIContext{}, flags, &config.MountConfig{})
+
+	if assert.Nil(t, err) {
+		assert.Equal(t, v.GcsConnection.ClientProtocol, cfg.Protocol("http1"))
+	}
+}
+
+func TestInvalidClientProtocol(t *testing.T) {
+	flags := &flagStorage{
+		ClientProtocol: mountpkg.ClientProtocol("http3"),
+	}
+
+	_, err := PopulateNewConfigFromLegacyFlagsAndConfig(&mockCLIContext{}, flags, &config.MountConfig{})
+
+	assert.NotNil(t, err)
```

Comment:
Can we assert on actual error in this case. I believe it's a custom error in this case.

Reply:
Usually, it's not a good idea to test errors for reasons outlined here - https://goto.google.com/go-style/decisions#test-error-semantics. Let me know if you think otherwise.

Reply:
Makes sense, let's add appropriate error checks for the API (decode hooks later).